File: src/main/java/be/vdab/movies/klanten/Klant.java
package be.vdab.movies.klanten;

public class Klant {
    private final long id;
    private final String familienaam;
    private final String voornaam;
    private final String straatNummer;
    private final int postcode;
    private final String gemeente;

    public Klant(long id, String familienaam, String voornaam, String straatNummer, int postcode, String gemeente) {
        this.id = id;
        this.familienaam = familienaam;
        this.voornaam = voornaam;
        this.straatNummer = straatNummer;
        this.postcode = postcode;
        this.gemeente = gemeente;
    }

    public long getId() {
        return id;
    }

    public String getFamilienaam() {
        return familienaam;
    }

    public String getVoornaam() {
        return voornaam;
    }

    public String getStraatNummer() {
        return straatNummer;
    }

    public int getPostcode() {
        return postcode;
    }

    public String getGemeente() {
        return gemeente;
    }
}

File: src/main/resources/static/js/klant.js
"use strict";
import {byId, toon, verberg, verwijderChildElementenVan} from "./util.js";

byId("zoek").onclick = async function () {
    verbergPizzasEnFouten();
    const woordInput = byId("woord");
    if (woordInput.checkValidity()) {
        findByWoord(woordInput.value);
    } else {
        toon("woordFout");
        woordInput.focus();
    }
};

function verbergPizzasEnFouten() {
    verberg("klantenTable");
    verberg("woordFout");
    verberg("storing");
    verberg("geenResultaten");
}

async function findByWoord(woord) {
    const response = await fetch(`klanten?naamBevat=${woord}`);
    if (response.ok) {
        const klanten = await response.json();
       // console.log(klanten)
        const klantenBody = byId("klantenBody");
        verwijderChildElementenVan(klantenBody);

        if (klanten.length === 0) {
            toon("geenResultaten");
            verberg("KlantenTable");
        } else {
            for (const klant of klanten) {
                toon("klantenTable");
                const tr = klantenBody.insertRow();
                const klantNaamVoornaamCell = tr.insertCell();
                klantNaamVoornaamCell.innerText = klant.familienaam + " " + klant.voornaam;
                klantNaamVoornaamCell.style.textDecoration = "underline";
                klantNaamVoornaamCell.onclick = function () {
                    window.location.href = 'bevestig.html?id=' + klant.id;
                    sessionStorage.setItem("klantStorage", JSON.stringify(klant));
                };

                tr.insertCell().innerText = klant.straatNummer;
                tr.insertCell().innerText = klant.postcode;
                tr.insertCell().innerText = klant.gemeente;
            }
        }
    } else {
        toon("storing");
    }
}